SimCalorimetry/EcalSimAlgos/src/EcalLiteDTUCoder.cc

```diff
@@ -0,0 +1,241 @@
+//Quando è selezionata l'opzione dei 16 samples:
+//5 samples prima, il sample saturo e poi 10 samples dopo
+const float DTUVersion = 1.0;
+
+EcalLiteDTUCoder::EcalLiteDTUCoder(bool addNoise,
+                             bool PreMix1,
+                             EcalLiteDTUCoder::Noisifier* ebCorrNoise0,
+                             EcalLiteDTUCoder::Noisifier* ebCorrNoise1)
+    : m_peds(nullptr),
+      m_gainRatios(nullptr),
+      m_intercals(nullptr),
+      //   m_maxEneEB    (      1668.3 ) , // 4095(MAXADC)*12(gain 2)*0.035(GeVtoADC)*0.97
+      m_maxEneEB(ecalPh2::maxEneEB),  // Maximum for CATIA: LSB gain 10: 0.048 MeV
+      m_addNoise(addNoise),
+      m_PreMix1(PreMix1)
+
+{
+  m_ebCorrNoise[0] = ebCorrNoise0;
+  assert(nullptr != m_ebCorrNoise[0]);
+  m_ebCorrNoise[1] = ebCorrNoise1;
+}
+
+EcalLiteDTUCoder::~EcalLiteDTUCoder() {}
+
+void EcalLiteDTUCoder::setFullScaleEnergy(double EBscale) {
+  //   m_maxEneEB = EBscale ;
+  m_maxEneEB = ecalPh2::maxEneEB;  //I don 't know where is setFullScaleEnergy first call
+}
+
+void EcalLiteDTUCoder::setPedestals(const EcalLiteDTUPedestalsMap* pedestals) { m_peds = pedestals; }
+
+void EcalLiteDTUCoder::setGainRatios(const EcalCATIAGainRatios* gainRatios) { m_gainRatios = gainRatios; }
+
+void EcalLiteDTUCoder::setIntercalibConstants(const EcalIntercalibConstantsMC* ical) { m_intercals = ical; }
+
+double EcalLiteDTUCoder::fullScaleEnergy(const DetId& detId) const {
+  //return detId.subdetId() == EcalBarrel ? m_maxEneEB : m_maxEneEE ;
+  return m_maxEneEB;
+}
+
+void EcalLiteDTUCoder::analogToDigital(CLHEP::HepRandomEngine* engine,
+                                    const EcalSamples& clf,
+                                    EcalDataFrame_Ph2& df) const {
+  df.setSize(clf.size());
+  encode(clf, df, engine);
+}
+
+void EcalLiteDTUCoder::encode(const EcalSamples& ecalSamples,
+                           EcalDataFrame_Ph2& df,
+                           CLHEP::HepRandomEngine* engine) const {
+  assert(nullptr != m_peds);
+
+  const unsigned int csize(ecalSamples.size());
+
+  DetId detId = ecalSamples.id();
+  double Emax = fullScaleEnergy(detId);
+
+  //....initialisation
+  if (ecalSamples[5] > 0.)
+    LogDebug("EcalLiteDTUCoder") << "Input caloSample"
+                              << "\n"
+                              << ecalSamples;
+
+  //N Gains set to 2 in the .h
+  double pedestals[ecalPh2::NGAINS];
+  double widths[ecalPh2::NGAINS];
+  //float  gains[ecalPh2::NGAINS];
+  double LSB[ecalPh2::NGAINS];
+  double trueRMS[ecalPh2::NGAINS];
+
+  double icalconst = 1.;
+  findIntercalibConstant(detId, icalconst);
+
+  for (unsigned int igain(0); igain < NGAINS; ++igain) {
+    // fill in the pedestal and width
+
+    findPedestal(detId, igain, pedestals[igain], widths[igain]);
+    //I insert an absolute value in the trueRMS
+    trueRMS[igain] = std::sqrt(std::fabs(widths[igain] * widths[igain] - 1. / 12.));
```

Comment:
use `std::abs()`
also, magic number 1/12 should be named constant

Reply:
sqrt(12) is indeed a magic number (rms of flat distribution)